src/renderers/webgl/WebGLState.js

```diff
@@ -551,6 +538,20 @@ function WebGLState( gl, extensions, utils ) {
 
 		}
 
+		var blendEquationCacheHit = false;
+
+		// Blending caching to remove unecessary WebGL calls
+		if ( currentMaterial != null ) {
+
+			if ( currentMaterial.lastBlendingSet == blending &&
+				currentMaterial.lastPremultipliedAlphaSet == premultipliedAlpha ) {
+
+			    blendEquationCacheHit = true;
+
+			}
+
+		}
+
 		if ( blending !== CustomBlending ) {
 
 			if ( blending !== currentBlending || premultipliedAlpha !== currentPremultipledAlpha ) {
```

Comment:
I was confused as well but it turns out that the currentBlending variable is shared between different materials so it did not ensure the blending equation caching. maybe it was supposed to serve another purpose that i'm not aware of, so i kept it just in case.

Reply:
I don't think the change is correct. Caching WebGL State information is something different than caching uniforms. The WebGL state is always related to a draw call, it can't retain information that is related to a material.

Reply:
Yes I believe you are right about that. I believe that the reason of the redundant calls was that the blending equation function is acually the same (gl.FUNC_ADD) for different blending modes and that probably was the reason why this PR seemed to work. Maybe there should be different caches for blending equations and functions to prevent the redundant calls. Any thoughts?

Reply:
Thanks for investigating this! 👍